libs/cmdio/exec.go

```diff
@@ -0,0 +1,162 @@
+package cmdio
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type Executor struct {
+	dir         string
+	scriptFiles []string
+}
+
+func NewCommandExecutor(dir string) *Executor {
+	return &Executor{
+		dir:         dir,
+		scriptFiles: nil,
+	}
+}
+
+func (e *Executor) StartCommand(ctx context.Context, command string) (*exec.Cmd, io.Reader, error) {
```

Comment:
Re: the signature -- do we need to expose `*exec.Cmd` here or can we narrow it to just the `Wait()` function?